src/pwncore/models/user.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from tortoise import fields
+from tortoise.exceptions import IntegrityError
+from tortoise.models import Model
+from tortoise.expressions import Q
+
+if TYPE_CHECKING:
+    from pwncore.models.container import Container
+
+__all__ = ("User", "Team", "generate_trigger_script")
+# Not too sure here
+
+
+class User(Model):
+    # Registration numbers and other identity tags
+    # abstractly just represents one person, expand this
+    # field for Identity providers
+    tag = fields.CharField(128, unique=True)
+    name = fields.CharField(255)
+    email = fields.TextField()
+    phone_num = fields.CharField(15)
+
+    team: fields.ForeignKeyNullableRelation[Team] = fields.ForeignKeyField(
+        "models.Team", "members", null=True, on_delete=fields.OnDelete.SET_NULL
+    )
+
+    async def save(self, *args, **kwargs):
+        # TODO: Insert/Update in one query
+        # Reason why we dont use pre_save: overhead, ugly
+        if self.team is not None:
+            cnt = await self.team.members.filter(~Q(id=self.pk)).count()
+            if cnt >= 3:
```

Comment:
fetch the maximum count from the config and please change variable name to `count` or anything else instead of `cnt` :cry:

Reply:
about the maximum count I need #5 to be merged, so that can be included in #5 or a new PR